gcc/rust/rust-session-manager.cc

```diff
@@ -113,24 +114,32 @@ infer_crate_name (const std::string &filename)
 static bool
 validate_crate_name (const std::string &crate_name, Error &error)
```

Comment:
this function would probably benefit from the new `tl::expected` type we have, but that's a problem for another PR!